### PR TITLE
add AWS plugin for AWS MSK's IAM access control 

### LINKED ISSUE #152: KafkaCtl with AWS MSK and IAM auth
Hello,

I'd like to use KafkaCtl with AWS MSK. The MSK cluster is currently enabled with IAM auth only. Is there a way to enable kafkactl to use IAM SASL auth?

This configuration is working for the kafka java commands:
```
security.protocol=SASL_SSL
sasl.mechanism=AWS_MSK_IAM
sasl.jaas.config=software.amazon.msk.auth.iam.IAMLoginModule required;
sasl.client.callback.handler.class=software.amazon.msk.auth.iam.IAMClientCallbackHandler
```

Thanks for your help!

### PR DESCRIPTION
**Pull request by KeisukeYamashita**
# Description

Implemented a token provider plugin using IAM access control to enable access to MSK without static credentials such as passwords. With this implementation, users can execute `kafkactl` with IAM access control, eliminating the need to manage IDs and passwords. 

Previously, as described in the issue below, operators and administrators had to enable `SASL/SCRAM` authentication or perform cumbersome operations due to `kafkactl` not supporting IAM access control for AWS MSK. This implementation simplifies the process and enhances security.

Fixes https://github.com/deviceinsight/kafkactl/issues/152

I would like to update the [deviceinsight/kafkactl](https://github.com/deviceinsight/kafkactl) documentation after this plugin gets approved and released.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Documentation

- [x] the change is mentioned in the `## [Unreleased]` section of `<plugin>/CHANGELOG.md`
- [x] the configuration options in `<plugin>/README.adoc` were updated

### CODE REVIEW

File: aws/main.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	}
+
+	params := parsedURL.Query()
+	d := params.Get("X-Amz-Date")
+	if d == "" {
+		return true, nil
+	}
+	signedAt, err := time.Parse("20060102T150405Z", d)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse the 'X-Amz-Date' param from signed url: %w", err)
```
Comment (KeisukeYamashita) [inline aws/main.go:123]: > [!NOTE]
> Unlike `ExpiresQueryKey`, this string is not exposed.
> Ref: https://github.com/aws/aws-msk-iam-sasl-signer-go/pull/19

Comment (d-rk) [reply aws/main.go:123]: :ok_hand:

**Comment by d-rk** [comment]
@KeisukeYamashita thank you again for the PR. I added some smaller adjustments, but in general everything looked fine :100: 

I will build a release for this, by the end of today hopefully.

**Comment by KeisukeYamashita** [comment]
@d-rk  Sorry for the late reply 🙏  Thank you very much for your review and the release 🚀

**Pull request merged**